src/OrchardCore.Modules/OrchardCore.Users/Controllers/ChangeEmailController.cs

```diff
@@ -38,7 +38,7 @@ public class ChangeEmailController : Controller
             _userService = userService;
             _userManager = userManager;
             _siteService = siteService;
-            _emailAddressValidator = emailAddressValidator ?? throw new ArgumentNullException(nameof(emailAddressValidator));
+            _emailAddressValidator = emailAddressValidator;
```

Comment:
I'm not sure why we don't check from `null` in constructor which is useful for `public` APIs, if the intend to prevent DI from throws, it will if the service is not there

Reply:
The DI container will throw a DI exception before it ever reaches this null check so a null check here is unreachable code

Reply:
How? I think the check need to be in services public APIs, not in controller as you said